configure.ac

```
@@ -466,12 +473,28 @@ AC_ARG_ENABLE(network,
               AS_HELP_STRING([--disable-network],
                              [Disable networking features (modem, ipx)]),,
               enable_network=yes)
-AC_CHECK_HEADER(SDL_net.h,have_sdl_net_h=yes,)
+AC_ARG_ENABLE(sdl_net_static,
+              AC_HELP_STRING([--enable-sdl-net-static],
+                             [Link SDL2_Net statically (not recommended)]),
+              enable_sdl_net_static=yes,
+              enable_sdl_net_static=no)
+AC_CHECK_HEADER(SDL_net.h, have_sdl_net_h=yes,)
 AC_CHECK_LIB(SDL2_net, SDLNet_Init, have_sdl_net_lib=yes, , )
 AC_MSG_CHECKING([whether networking features will be enabled])
 if test x$enable_network = xyes ; then
   if test x$have_sdl_net_lib = xyes -a x$have_sdl_net_h = xyes ; then
-    LIBS="$LIBS -lSDL2_net"
+    if test x$enable_sdl_net_static = xyes ; then
+      case "$host" in
+        *-*-darwin*)
+          LIBS="$LIBS /usr/local/lib/libSDL2_net.a"
+          ;;
+        *)
+          LIBS="$LIBS -Wl,-Bstatic -lSDL2_net -Wl,-Bdynamic"
```

```suggestion
          LIBS="$LIBS -Wl,-Bstatic -lSDL2_net -lSDL2 -Wl,--no-undefined -lm -ldl -lpthread -lrt -Wl,-Bdynamic"
```
As these are flags suggested by `pkg-config --static --libs SDL2_net`.

Yes, this means that `--enable-sdl-net-static` implies `--enable-sdl-static` in practice. Maybe there's a better way to handle it. I am also ok with leaving it as it is for now - if it works on macOS.

---

I think there's a better way: `--enable-sdl-static` should throw that lever for SDL in general (and now we finally have the underlying mechanism to honor it for SDL2_Net as well).  Let me try again.

---

**./configure --enable-png-static**

``` text
checking png.h usability... yes
checking png.h presence... yes
checking for png.h... yes
checking for png_get_io_ptr in -lpng... yes
checking whether screenshots will be enabled... yes (static)
```

**./configure  --enable-sdl-static**

``` text
checking SDL_net.h usability... yes
checking SDL_net.h presence... yes
checking for SDL_net.h... yes
checking for SDLNet_Init in -lSDL2_net... yes
checking whether networking features will be enabled... yes (static)
```

That's looking better than the original commit, and saner behavior with SDL being static overall or not, and only having to throw one switch.  If someone finds a scenario were we need to tease these apart again, then it we can look into splitting the SDL switches again.  Although hopefully by then, we'll be on cmake and it can be smart enough to fallback to dynamic libs whenever a static one isn't found :-)